File: app/src/main/java/closer/vlllage/com/closer/handler/SystemSettingsHandler.java
package closer.vlllage.com.closer.handler;

import android.content.Intent;
import android.net.Uri;
import android.provider.Settings;

import closer.vlllage.com.closer.pool.PoolMember;

public class SystemSettingsHandler extends PoolMember {
    public void showSystemSettings() {
        Intent intent = new Intent();
        intent.setAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
        Uri uri = Uri.fromParts("package", $(ApplicationHandler.class).getApp().getPackageName(), null);
        intent.setData(uri);
        $(ActivityHandler.class).getActivity().startActivity(intent);
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/SystemSettingsHandler.kt
package closer.vlllage.com.closer.handler.helpers

import android.content.Intent
import android.net.Uri
import android.provider.Settings

import com.queatz.on.On

class SystemSettingsHandler constructor(private val on: On) {
    fun showSystemSettings() {
        val intent = Intent()
        intent.action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
        val uri = Uri.fromParts("package", on<ApplicationHandler>().app.packageName, null)
        intent.data = uri
        on<ActivityHandler>().activity!!.startActivity(intent)
    }
}
